app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt

```diff
@@ -314,35 +314,41 @@ class MainTextDrawableFragment : BaseFragment() {
         return inflater.inflate(R.layout.fragment_main_text, container, false)
     }
 
-    class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
+    inner class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
         override fun doInBackground(vararg params: Void?): Boolean? {
-            return (!StorageUtils.checkIfFilePresent(filename))
+            return when (StorageUtils.checkIfFilePresent(filename)) {
+                true -> {
```

Comment:
Please revert StoreAsync to its original state. This is messed up, as far as I know this logic will not work